optuna/storages/_journal/storage.py

```diff
@@ -29,6 +32,8 @@
 _logger = optuna.logging.get_logger(__name__)
 
 NOT_FOUND_MSG = "Record does not exist."
+# A heuristic interval number to dump snapshots
+SNAPSHOT_INTERVAL = 100
```

Comment:
How about making this parameter configurable by users?

By the current design of `BaseJournalLogSnapshot`, the all studies in the storage are serialized each time this interval is passed. The cost could not be negligible if there are too many studies and the evaluation time of each objective function call is too short (and the number of trials is too large).
If this parameter is configurable, increasing the value could be a workaround for such a case I think.